django/db/migrations/executor.py

```diff
@@ -199,9 +204,27 @@ def detect_soft_applied(self, project_state, migration):
                     # We have to fetch the model to test with from the
                     # main app cache, as it's not a direct dependency.
                     model = global_apps.get_model(model._meta.swapped)
+                if model._meta.proxy or not model._meta.managed:
+                    continue
                 if model._meta.db_table not in self.connection.introspection.table_names(self.connection.cursor()):
                     return False, project_state
                 found_create_migration = True
-        # If we get this far and we found at least one CreateModel migration,
+
+            elif isinstance(operation, migrations.AddField):
+                model = apps.get_model(migration.app_label, operation.model_name)
+                if model._meta.swapped:
+                    # We have to fetch the model to test with from the
+                    # main app cache, as it's not a direct dependency.
+                    model = global_apps.get_model(model._meta.swapped)
+                if model._meta.proxy or not model._meta.managed:
+                    continue
+
+                table = model._meta.db_table
+                db_field = model._meta.get_field(operation.name).column
+                fields = self.connection.introspection.get_table_description(self.connection.cursor(), table)
+                if db_field not in [f.name for f in fields]:
```

Comment:
I think the current version is clear. You could change the `[...]` to `(...)` though